src/linkml_transformer/transformer/object_transformer.py

```diff
@@ -110,8 +111,16 @@ def transform(
                     }
                 else:
                     do = dynamic_object(source_obj, sv, source_type)
+                    ctxt_obj = do
                     ctxt_dict = vars(do)
-                v = eval_expr(slot_derivation.expr, **ctxt_dict, NULL=None)
+
+                try:
+                    v = eval_expr(slot_derivation.expr, **ctxt_dict, NULL=None)
+                except Exception:
+                    aeval = Interpreter(usersyms={"src": ctxt_obj, "target": None})
+                    aeval(slot_derivation.expr)
+                    v = aeval.symtable["target"]
```

Comment:
This sets up the context object as the variable `src` and gets the slot value from the variable `target`. Example transform:
```py
d_test = [x.important_event_date for x in src.has_important_life_events if str(x.event_name) == "PASSED_DRIVING_TEST"]
target = d_test[0] if d_test else None
```
If you use `**ctxt_dict`, it is easy to accidentally have a variable with the same name as python function/keyword, hence restricting everything to be under `src`.